UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c

```diff
@@ -594,18 +633,35 @@ PlatformBootManagerBeforeConsole (
   //
   // Map ESC to Boot Manager Menu
   //
-  Esc.ScanCode    = FixedPcdGet16(PcdSetupMenuKey);;
+  Esc.ScanCode    = FixedPcdGet16(PcdSetupMenuKey);
   Esc.UnicodeChar = CHAR_NULL;
   EfiBootManagerGetBootManagerMenu (&BootOption);
   EfiBootManagerAddKeyOptionVariable (NULL, (UINT16) BootOption.OptionNumber, 0, &Esc, NULL);
 
   //
   // Map F12 to Boot Device List menu
   //
+  DEBUG((DEBUG_INFO, "KZM_PlatformBootManagerBeforeConsole: before start\n"));
   F12.ScanCode    = FixedPcdGet16(PcdBootMenuKey);
   F12.UnicodeChar = CHAR_NULL;
   OptionNumber    = GetBootManagerMenuAppOption ();
-  EfiBootManagerAddKeyOptionVariable (NULL, (UINT16)OptionNumber, 0, &F12, NULL);
+  EfiBootManagerAddKeyOptionVariable (&BootManagerMenuOption_delete, (UINT16)OptionNumber, 0, &F12, NULL);
+  DEBUG((DEBUG_INFO, "KZM_PlatformBootManagerBeforeConsole: after execution\n"));
+
+  VarSize = sizeof (BootMenuEnable);
+  Status = gRT->GetVariable (
+          L"BootManagerEnabled",
+          &gDasharoSystemFeaturesGuid,
+          NULL,
+          &VarSize,
+          &BootMenuEnable
+        );
+  if ((Status != EFI_NOT_FOUND) && (VarSize == sizeof(BootMenuEnable))) {
+    if (!BootMenuEnable) {
+      EfiBootManagerDeleteKeyOptionVariable(&BootManagerMenuOption_delete, (UINT16)OptionNumber, 0, &F12);

```

Comment:
```suggestion
      EfiBootManagerDeleteKeyOptionVariable(&BootManagerMenuOption_delete, 0, &F12);
```

The function doesn't expect an OptionNumber parameter, but because it takes a VA_LIST as arguments it doesn't throw an error. Instead it takes the value and parses it as a key modifier.

With this change it starts to work.  Will push a change shortly and clean up the debug stuff.

Reply:
done